src/command_modules/azure-cli-consumption/azure/cli/command_modules/consumption/commands.py

```diff
@@ -28,8 +37,28 @@ def load_command_table(self, _):
 
     with self.command_group('consumption reservations details') as d:
         d.custom_command('list', 'cli_consumption_list_reservations_details', transform=transform_reservation_details_list_output,
-                         exception_handler=consumption_exception_handler, validator=None, client_factory=reservations_details_mgmt_client_factory)
+                         exception_handler=consumption_exception_handler, client_factory=reservations_details_mgmt_client_factory)
 
     with self.command_group('consumption pricesheet') as p:
         p.custom_command('show', 'cli_consumption_list_pricesheet_show', transform=transform_pricesheet_show_output,
-                         exception_handler=consumption_exception_handler, validator=None, client_factory=pricesheet_mgmt_client_factory)
+                         exception_handler=consumption_exception_handler, client_factory=pricesheet_mgmt_client_factory)
+
+    with self.command_group('consumption marketplace') as p:
+        p.custom_command('list', 'cli_consumption_list_marketplace', transform=transform_marketplace_list_output,
+                         exception_handler=consumption_exception_handler, validator=validate_both_start_end_dates, client_factory=marketplace_mgmt_client_factory)
```

Comment:
Generally, `validate_both_start_end_dates` would be a parameter level validator, not a command level validator. Unless you need to specifically sequence validators, there is no good reason to use a command-level validator.

Reply:
moved